src/reflect/scala/reflect/internal/Chars.scala

```diff
@@ -118,6 +118,12 @@ trait Chars {
          '|' | '/' | '\\' => true
     case c => isSpecial(c)
   }
+
+  def isBiDiCharacter(c: Char): Boolean = (c: @switch) match {
+    case '\u202a' | '\u202b' | '\u202c' | '\u202d' | '\u202e' |
+         '\u2066' | '\u2067' | '\u2068' | '\u2069' => true
```

Comment:
I think we should forbid all 25 characters with the Unicode property `Bidi_Control`, as defined in `PropList.txt`:
https://www.unicode.org/Public/UCD/latest/ucd/PropList.txt

The [original vulnerability report](https://trojansource.codes/trojan-source.pdf) only mentions those 9 code points in Table I, sure, but it does refer to the Unicode specification for "the Complete list", and that goes to https://www.unicode.org/reports/tr9/tr9-42.html#Directional_Formatting_Characters, which says that all the involved characters have the property `Bidi_Control`.

Reply:
👍  -- though they are 12, not 25, right?

Reply:
@sjrd maybe we were a bit too eager here? The paper says "Table I provides a list of Bidi override characters relevant to this attack". I guess by "complete list" they meant "complete list of directional formatting characters", not "complete list of characters that can be exploited".

The community build found a few `U+200E` "left-to-right mark" characters here (accidental I assume, they don't serve any purpose): https://github.com/monix/monix/blob/ec5555c28406b21f785e7141970765c729988810/monix-eval/shared/src/main/scala/monix/eval/Coeval.scala#L40. They are not flagged by GitHub, for example. VS Code screenshot:

<img width="877" alt="image" src="https://user-images.githubusercontent.com/119636/167397751-8a01d614-2021-4c98-b8a5-d0f6b8c66cc2.png">

There's an example on Wikipedia showing where such marks are needed: https://en.wikipedia.org/wiki/Bidirectional_text#Marks. Could be relevant when using other alphabets in scaladoc comments.

Reply:
Ah yes, you are right, indeed.